server/game/gameSystems/DrawSystem.ts

```diff
@@ -38,5 +39,23 @@ export class DrawSystem<TContext extends AbilityContext = AbilityContext> extend
         const { amount } = this.generatePropertiesFromContext(context, additionalProperties);
         super.addPropertiesToEvent(event, player, context, additionalProperties);
         event.amount = amount;
+        // If there are not enough cards left in deck to draw the specified amount, instead draw the remainder and take damage to base equal to thrice the difference.
+        event.checkCondition = () => {
+            if (event.cancelled || event.resolved || event.name === EventName.Unnamed) {
+                return;
+            }
+            if (!event.condition(this)) {
+                event.cancel();
+            }
+            if (event.amount > event.player.drawDeck.length) {
```

Comment:
Will discuss in chat, but I think this logic needs to be in `eventHandler`

Reply:
So how exactly should the damage be done? I'd think it needs to be a damage event, rather than just directly adding damage in DrawSystem.eventHandler, for consistency and futureproofing(potential triggers/replacement effects for damage on base) right? Should it be a subevent? A new event window?

Reply:
So, looking at it, I think it might actually be a replacement effect? It uses the word "instead" which is one of the words indicating replacement.

My first thought would then be to use the replacement mechanism to check if the draw event will overshoot the number of cards left in deck, then if so, optionally create a new draw event with the right draw number (if there are any cards at all to draw) and a damage event.

Probably the easiest way to create both is to just have the `replacementImmediateEffect` parameter of `ReplacementEffectSystem` be a `simultaneous()` with the two events. I _think_ that would work but we've never tried it before so there might be some hiccups. We may need to add a new solution for how to have replacementEffect create multiple events.

Reply:
Actually, I thought about this a little more - I guess replacement effect isn't the right way to do it since that kind of implies that there is an outside effect that is altering the normal resolution. But creating the damage events is part of the normal resolution of draw. So then I think it's probably a contingent event.

Reply:
Apologies for all the back-and-forth on this, more complicated than I was expecting. Thanks again for tackling it.

Reply:
No worries. I considered several different implementations myself before I got to where I submitted the PR. Just checking: does making it a contingent event imply, as far as the game engine is concerned, that the original draw event happened successfully?

Reply:
Sort of - it's more the reverse, the game engine won't create the contingent event until it's sure the source event will resolve.

Overwhelm is a good example - the overwhelm event is created as a contingent event off the attacker damage, but only after the replacement effect step because shield might cancel the attacker event.

Reply:
Right. I'm concerned(probably unnecessarily) that having a false draw event(no cards actually drawn) resolve could be bad in the future if there's such a thing as triggering on draw. For now though I've coded the contingent event implementation and tests are passing.

Reply:
This is a valid concern (if I understand right) and I think I have an answer for this, messaged in discord to continue the discussion